src/main/scala/chiseltest/simulator/jna/JNAUtils.scala

```diff
@@ -0,0 +1,145 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chiseltest.simulator.jna
+import com.sun.jna._
+import scala.collection.JavaConverters
+
+object JNIUtils {
+  def javaHome: String = System.getProperty("java.home") match {
+    case s: String if s.endsWith("/jre") => s.dropRight(4)
+    case s: String                       => s
+  }
+  def osIncludeName: String = System.getProperty("os.name") match {
+    case "Mac OS X" => "darwin"
+    case "Linux"    => "linux"
+    case s: String => s
+  }
+
+  /** additional ccFlags that are required in order to compile into a library that can be loaded with JNI */
+  def ccFlags: Seq[String] = Seq(
+    "-fPIC",
+    "-shared",
+    s"-I$javaHome/include",
+    s"-I$javaHome/include/$osIncludeName",
+    // hide all symbols in the shared object by default
+    // https://stackoverflow.com/questions/435352/limiting-visibility-of-symbols-when-linking-shared-libraries
+    "-fvisibility=hidden"
+  )
+  def ldFlags: Seq[String] = Seq(
+    "-shared",
+    "-dynamiclib",
+    // hide all symbols in the shared object by default
+    // https://stackoverflow.com/questions/435352/limiting-visibility-of-symbols-when-linking-shared-libraries
+    "-fvisibility=hidden"
+  )
+
+  /** needs to match up with [[TesterSharedLibInterface]]! */
+  val Methods = Seq(
+    ("void", "step", Seq()),
+    ("void", "update", Seq()),
+    ("void", "finish", Seq()),
+    ("void", "resetCoverage", Seq()),
+    ("void", "writeCoverage", Seq("filename" -> "String")),
+    ("void", "poke", Seq("id" -> "int", "value" -> "long")),
+    ("long", "peek", Seq("id" -> "int")),
+    ("void", "poke_wide", Seq("id" -> "int", "offset" -> "int", "value" -> "long")),
+    ("long", "peek_wide", Seq("id" -> "int", "offset" -> "int"))
+  )
+
+  private var idCounter = 123
+  private def getUniqueId: Int = synchronized {
+    val id = idCounter
+    idCounter += 1
+    id
+  }
+
+  def compileAndLoadJNIClass(libPath: os.Path, className: String): TesterSharedLibInterface = {
```

Comment:
className arg is unused
```suggestion
  def compileAndLoadJNIClass(libPath: os.Path): TesterSharedLibInterface = {
```
Making this change requires a change in VerilatorSimulator.scal:140